include/dynd/kernels/string_split_kernel.hpp

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (C) 2011-15 DyND Developers
+// BSD 2-Clause License, see LICENSE.txt
+//
+
+// String split kernel
+
+#pragma once
+
+#include <dynd/string.hpp>
+#include <dynd/string_search.hpp>
+#include <dynd/types/var_dim_type.hpp>
+
+namespace dynd {
+  namespace nd {
+
+    struct string_split_kernel
+      : base_kernel<string_split_kernel, 2> {
+
+      typedef string_split_kernel self_type;
+
+      memory_block_data *m_dst_memblock;
+
+      string_split_kernel(memory_block_data *dst_memblock) :
+        m_dst_memblock(dst_memblock) {
+
+      }
+
+      static void instantiate(char *DYND_UNUSED(static_data), char *DYND_UNUSED(data),
+                              kernel_builder *ckb, const ndt::type &DYND_UNUSED(dst_tp),
+                              const char *dst_arrmeta, intptr_t DYND_UNUSED(nsrc),
+                              const ndt::type *DYND_UNUSED(src_tp),
+                              const char *const *DYND_UNUSED(src_arrmeta),
+                              kernel_request_t kernreq, intptr_t DYND_UNUSED(kwd),
+                              const nd::array *DYND_UNUSED(kwds),
+                              const std::map<std::string, ndt::type> &DYND_UNUSED(tp_vars)) {
+        ckb->emplace_back<self_type>(
+          kernreq, reinterpret_cast<const ndt::var_dim_type::metadata_type *>(dst_arrmeta)->blockref.get());
+      }
+
+      void single(char *dst, char *const *src) {
+        ndt::var_dim_type::data_type *dst_v = reinterpret_cast<ndt::var_dim_type::data_type *>(dst);
+
+        const string *const *s = reinterpret_cast<const string *const *>(src);
+        const string &haystack = *(s[0]);
+        const string &needle = *(s[1]);
+
+        intptr_t count = dynd::string_count(haystack, needle);
+
```

Comment:
There is an ambiguity between the regular function in `dynd::string_count` and the elwise kernel in `dynd::nd::string_count`.  In this case I wan't to use the former, but C++ defaults to the latter, since this block is in `dynd::nd`.  I can rename things if that would help...